plumbing/transport/registry.go

```diff
@@ -0,0 +1,39 @@
+package transport
+
+import (
+	"fmt"
+	"sync"
+)
+
+// registry are the protocols supported by default.
+var (
+	registry = map[string]Transport{}
+	mtx      sync.Mutex
+)
+
+// Register adds or modifies an existing protocol.
```

Comment:
Where the API changed substantially, I think we should either add comments to clarify that, or find another place to amass all the API changes, so it is easier for users to migrate to V6.

```suggestion
// Register adds or modifies an existing protocol.
// Equivalent to client.InstallProtocol in go-git before V6.
```